docs-src/docs/guides/terraform_beyond_layer0.md

```diff
@@ -0,0 +1,519 @@
+  count = "${length(var.environments)}"
+
+  name = "${lookup(var.environments, count.index)}_demo"
+}
+```
+
+Let's have a more in-depth look in how this works. You can start buy navigating to `./terraform-beyond-layer0/example-2' folder. Start by running the plan command.
+
+`terraform plan`
+
+Outputs:
+```
++ module.environment.aws_dynamodb_table.guestbook.0
+    ...
+    name:                      "dev_guestbook"
+...
++ module.environment.aws_dynamodb_table.guestbook.1
+    ..
+    name:                      "staging_guestbook"
+...
+```
+
+Note that you will a copy of each resource for each environment specified in your environments file in `./example-2/variables.tf`. Go ahead and run apply.
+
+`terraform apply`
+
+Outputs:
+```
+Apply complete! Resources: 10 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+guestbook_urls = 
+<dev_url>
+<staging_url>
+```
+
+You have now created two separate environments using a single terraform configuration: dev & staging. You can navigate to both the urls output and you should note that they are separate instances of the guestbook application backed with their own separate data store.
+
+A common use case for maintaining different environments is configure each environment slightly differently. For example, might want to scale your Layer0 service to 3 for staging and leave it as 1 for the dev environment. This can be done easily by using conditional logic to set our `scale` parameter in the layer0 service configuration in `./example-2/main.tf`. Go ahead and open `main.tf` in a text editor. Navigate to the `layer0_service guestbook` section. Uncomment the scale parameter so that your configuration looks like below.
+
+```
+resource "layer0_service" "guestbook" {
+  count = "${length(var.environments)}"
+
+  name          = "${element(layer0_environment.demo.*.name, count.index)}_guestbook_svc"
+  environment   = "${element(layer0_environment.demo.*.id, count.index)}"
+  deploy        = "${element(layer0_deploy.guestbook.*.id, count.index)}"
+  load_balancer = "${element(layer0_load_balancer.guestbook.*.id, count.index)}"
+  scale         = "${lookup(var.service_scale, lookup(var.environments, count.index), "1")}"
+}
+```
+
+The variable `service_scale` is already defined in `variables.tf`. If you now go ahead and run plan, you will see that the `guestbook` service for only the `staging` environment will be scaled up.
+
+`terraform plan`
+
+Outputs:
+
+```
+~ layer0_service.guestbook.1
+    scale: "1" => "3"
+```
+
+The downside of this approach however is that all your environments are using the same state file. Ideally you would want each environment to be completely isolated from other environments, which they mostly are. But sharing a state file breaks some of this encapsulation. Should there ever be a situation where your state file becomes corrupt, it would affect your ability to service all the environments till you resolve the issue by potentially rolling back to a previous copy of the state file. 
```

Comment:
> Ideally you would want each environment to be completely isolated from other environments, which they mostly are. But sharing a state file breaks some of this encapsulation.

I'd remove this sentence. It's conflating the idea of networking encapsulation (the "selling point" of Layer0 environments), with terraform resource management, and users probably won't be able to spot the difference.